x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/LearningToRankConfigTests.java

```diff
@@ -140,6 +148,66 @@ public void testDuplicateFeatureNames() {
         expectThrows(IllegalArgumentException.class, () -> builder.build());
     }
 
+    public void testLicenseSupport_ForPutAction_RequiresEnterprise() {
+        var config = randomLearningToRankConfig();
+        AtomicInteger currentTime = new AtomicInteger(100); // non zero start time
+
+        var licenseTestValues = Map.of(
+            License.OperationMode.TRIAL,
+            true,
+            License.OperationMode.BASIC,
+            false,
+            License.OperationMode.STANDARD,
+            false,
+            License.OperationMode.GOLD,
+            false,
+            License.OperationMode.PLATINUM,
+            false,
+            License.OperationMode.ENTERPRISE,
+            true
+        );
+
+        for (Map.Entry<License.OperationMode, Boolean> entry : licenseTestValues.entrySet()) {
+            var licenseStatus = new XPackLicenseStatus(entry.getKey(), true, "");
+            var licenseState = new XPackLicenseState(currentTime::get, licenseStatus);
+            assertThat(
+                Strings.format("Expected license state for [%s] to be [%s]", entry.getKey(), entry.getValue()),
+                config.isLicenseAllowedForAction(RestRequest.Method.PUT, licenseState),
+                is(entry.getValue())
+            );
+        }
+    }
+
+    public void testLicenseSupport_ForGetAction_RequiresPlatinum() {
+        var config = randomLearningToRankConfig();
+        AtomicInteger currentTime = new AtomicInteger(100); // non zero start time
+
+        var licenseTestValues = Map.of(
```

Comment:
> I think it would be better to use some of the built in LicenseUtil methods

I like the idea of not having to maintain the license map there... but I'm confused a bit... the `LicenseUtils` class you are utilizing in your tests is specific to the `enterprise-search` plugin, no?

Reply:
You're right that Ent search has it's own version of `LicenseUtils`. 🤦‍♀️ 

I was originally thinking of [LicenseUtils](https://github.com/elastic/elasticsearch/blob/9b1f07aa12a1bae60090ff162e257b36294dbda6/x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseUtils.java#L25) which is in core, which has some convenience methods. 

But it looks like you can just copy the same approach that Ent search uses and define the license along the lines of: 

```
LicensedFeature.momentary(
        null,
        XPackField.<whatever inference uses>,
        License.OperationMode.ENTERPRISE
    );
```